design/plankton/CommunicationLayerDesign.md

```diff
@@ -190,7 +190,21 @@ I believe this design will satisfy all the listed requirements. But I would be h
 * The design lacks a good way to enforce that a given device-adapter class uses any shared-code.
 * Can we implement `EnvironmentAdapters` to run on the same thread as the rest of the application? (I think the answer is probably yes). If not locking may become a problem.
 
+#### New package structure
 
+With the design explained above it becomes more straight forward to specify a package structure that is suitable for distribution (e.g. as a package in PyPI). What we could aim for is a package that has several modules:
+
+* `core`: Contains the state machine and base classes for different adapters, as well as some behavior that is useful for devices (e.g. "approach a value" stuff discussed in [#81](https://github.com/DMSC-Instrument-Data/plankton/issues/81)). These could be in sub-modules/packages `statemachine`, `behavior`, `adapters`, `utilities`.
+* `devices`: These would contain one sub-module per device, at the moment `chopper` and `linkam_t95`, which in turn contain the device implementation as well as the adapter(s) associated with that device. The setups should also be in a sub-module of the device.
+* `devices_contrib` (maybe there is a better name?): This is where external developers would add their devices to begin with. As @owenarnold wrote above, later refactoring done by us (or experienced developers) would move the device into the `devices` package. There is no strict need to have an entire directory per device, it could be a single Python file to begin with.
```

Comment:
I'd say once someone is involved enough, they could very well clone the git repository and put their file in the right place. We should definitely put together a "how to contribute" document, but with explanations our "target audience" should be able to do it.

If not, you are right, there should be another channel for contributions. But for the time being it might be enough to be reachable via e-mail?